recipe/build.sh

```diff
@@ -1,6 +1,13 @@
 if [[ "$target_platform" == "osx-64" ]]; then
-    # Remove when https://github.com/scikit-build/scikit-build/pull/441 lands
-    sed -i.bak "s/'-DSLYCOT_VERSION/'-DCMAKE_OSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET', '-DSLYCOT_VERSION/g" setup.py
+  # Remove when https://github.com/scikit-build/scikit-build/pull/441 lands
+  sed -i.bak "s/'-DSLYCOT_VERSION/'-DCMAKE_OSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET', '-DSLYCOT_VERSION/g" setup.py
+
+  # Fix a bug in gfortran 4. (This is fixed in gfortran 7)
+  TOOLS_DIR=$(dirname $($FC --print-libgcc-file-name))
+  if [[ ! -f "$TOOLS_DIR/ld" ]]; then
+    ln -sf $LD $TOOLS_DIR/ld
+    ln -sf $LD $BUILD_PREFIX/bin/ld
+  fi
```

Comment:
I wonder if we could ship a patched gfortran recipe from upstream with this? We could create a shell package that installs gfortran and applies the patch on activation

Reply:
I'd rather that gfortran 4 be dropped.